server/src/main/java/umm3601/user/UserController.java

```diff
@@ -73,44 +75,57 @@ public void getUser(Context ctx) {
     }
   }
 
-  /**
-   * Delete the user specified by the `id` parameter in the request.
-   *
-   * @param ctx a Javalin HTTP context
-   */
-  public void deleteUser(Context ctx) {
-    String id = ctx.pathParam("id");
-    userCollection.deleteOne(eq("_id", new ObjectId(id)));
-  }
-
   /**
    * Get a JSON response with a list of all the users.
    *
    * @param ctx a Javalin HTTP context
    */
   public void getUsers(Context ctx) {
+    Bson combinedFilter = constructFilter(ctx);
+    Bson sortingOrder = constructSortingOrder(ctx);
+
+    // All three of the find, sort, and into steps happen "in parallel" inside the
```

Comment:
For my own benefit, could I ask about what "in parallel" means here? (Thanks! 😃)

Reply:
I wasn't thrilled about my use of "in parallel" (which was part of the use of the scare quotes, I suspect).

What I'm trying to get at is that MongoDB doesn't so the whole search, and _then_ sort the results, and _then_ put them in an array. Instead there's some streaming/lazy evaluation stuff going on. I don't don't if (a) that's particularly important or (b) I'm conveying that in a way that is remotely useful. Thoughts on better wording?